client/src/components/EditProfile/EditProfileForm.tsx

```diff
@@ -0,0 +1,165 @@
+import { Formik, Form } from 'formik';
+import * as yup from 'yup';
+import { useEffect, useCallback, useState } from 'react';
+import { Container, Grid, Typography, Paper, MenuItem } from '@material-ui/core';
+import ButtonWrapper from './InputControls/ButtonWrapper';
+import useStyles from './useStyles';
+import TextFieldWrapper from './InputControls/TextFieldWrapper';
+
+interface formState {
+  firstName: string;
+  lastName: string;
+  gender: string;
+  birthday: string;
+  email: string;
+  phoneNumber: string;
+  location: string;
+  description: string;
+}
+const user = {
+  firstName: 'Rihanna',
+  lastName: 'Robyn',
+  gender: 'FEMALE',
+  birthday: '1991-03-01',
+  email: 'riro@gmail.com',
+  phoneNumber: '22384893879',
+  location: 'Bridgetown, Barbados',
+  description: 'I have been a taking care of pets ever since I was a child. I have great passion for what i do!',
+};
+const txtFieldsProps = [
+  { title: 'First name', fieldName: 'firstName', placeholder: 'John' },
+  { title: 'Last name', fieldName: 'lastName', placeholder: 'Doe' },
+  { title: 'Gender', fieldName: 'gender', placeholder: '' },
+  { title: 'Birthday', fieldName: 'birthday', placeholder: '' },
+  { title: 'Email address', fieldName: 'email', placeholder: 'john-doe@gmail.com' },
+  { title: 'Phone Number', fieldName: 'phoneNumber', placeholder: 'Phone' },
+  { title: 'Where you live', fieldName: 'location', placeholder: 'Location' },
+  { title: 'Describe yourself', fieldName: 'description', placeholder: 'About you' },
+];
+const genderCategories = [
+  { id: 1, option: 'MALE' },
+  { id: 2, option: 'FEMALE' },
+  { id: 3, option: 'OTHER' },
+];
+const formValidation = yup.object().shape({
+  firstName: yup.string().required('Required'),
+  lastName: yup.string().required('Required'),
+  gender: yup.string().required('Required'),
+  birthday: yup.string().required('Required'),
+  email: yup.string().email('Invalid email').required('Required'),
+  phoneNumber: yup
+    .number()
+    .integer()
+    .positive('Please enter a valid phone number')
+    .typeError('Please enter a valid phone number')
+    .required('Required'),
+  location: yup.string().required('Required'),
+  description: yup.string().required('Required'),
+});
+
+const EditProfileForm = (): JSX.Element => {
+  const classes = useStyles();
+  const { formContainer, formCaption, padding, label, formInput, submitBtn } = classes;
+  const [initialFormState, setInitialFormState] = useState<formState>({
+    firstName: '',
+    lastName: '',
+    gender: '',
+    birthday: '1994-03-01',
+    email: '',
+    phoneNumber: '',
+    location: '',
+    description: '',
+  });
+
+  const upDateProfileDetails = useCallback((newValues): void => {
+    fetch('localhost:3001/profile/update', {
```

Comment:
It would be better to take a relative path approach, maybe a .env file, than hardcoding 'localhost', as later it would be a headache